usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/MedicationManagerClientTest.java

```diff
@@ -0,0 +1,150 @@
+                + "{\n"
+                + "    \"date\": \"2020-02-13T10:30:00\",\n"
+                + "    \"name\" : \"Ibuprofeno\", \n"
+                + "    \"body\": {\n"
+                + "      \"quantity\": \"0.0\",\n"
+                + "      \"duration\": \"2\",\n"
+                + "      \"periodicity\": \"3\" \n"
+                + "    }\n"
+                + "  }"
+                + "   \n"
+                + "]");
+    }
+
+    @Test
+    public void createMedication() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(STATUS_OK);
+
+        int responseCode = client.createMedication(ACCESS_TOKEN, owner, petName, Medication);
+        assertEquals("Should return response code 200", STATUS_OK_INT, responseCode);
+    }
+
+    @Test
+    public void deleteByDateName() throws ExecutionException, InterruptedException {
+
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(STATUS_OK);
+
+        int responseCode = client.deleteByDateName(ACCESS_TOKEN, owner, petName, date, name);
+        assertEquals("Should return response code 200", STATUS_OK_INT, responseCode);
+    }
+
+    @Test
+    public void deleteAllMedications() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(STATUS_OK);
+
+        int responseCode = client.deleteAllMedications(ACCESS_TOKEN, owner, petName);
+        assertEquals("Should return response code 200", STATUS_OK_INT, responseCode);
+    }
+
+    @Test
+    public void getMedicationData() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(jsonMedicationData);
+
+        MedicationData response = client.getMedicationData(ACCESS_TOKEN, owner, petName, date, name);
+        assertEquals("Should return Medication data", response, MedicationData);
+    }
+
+    @Test
+    public void getAllMedicationData() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(jsonAllMedications);
+
+        List<Medication> response = client.getAllMedicationData(ACCESS_TOKEN, owner, petName);
+        assertEquals("Should return a Medication data list", response, MedicationList);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "Should return a Medication data list" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5403182)